content/WebAuthn/WebAuthn_Developer_Guide/Overview.adoc

```diff
@@ -1,55 +1,92 @@
+FIDO2 is built on top of two open standards: the Web Authentication API (WebAuthn) and the Client to Authenticator Protocol (CTAP2). The two work together and are required to achieve a strong authentication experience. The earlier FIDO U2F protocol working with external authenticators is now renamed CTAP1 in the WebAuthn specifications. FIDO2 and WebAuthn are backwards compatible with U2F authenticators.
 
+== How it works
+FIDO2/WebAuthn authentication consists of a registration ceremony and an authentication ceremony.
 
-==== On the Front End
+=== Registration Ceremony
+Assuming a user has already created an account with the online service, the user registers one or more authenticators against that identity. The authenticator could be a mobile phone, laptop, or YubiKey. Any authenticator that is compliant with CTAP1 or CTAP2 will work.
 
-To experience registration and authentication from the end user’s perspective, check out https://demo.yubico.com/webauthn[this demo].
+During the registration ceremony the authenticator creates a FIDO2 credential by generating a private key and public key. The private key stays in the hardware’s secure element while the public key is given to the online service. Possession of this authenticator in combination with a gesture to unlock the use of the credential is now proof of your identity. Unlike a password which is sent over the network and stored on a remote server, neither the private key nor the gesture ever leave the authenticator. They are not sent over the network nor stored on a remote server. Only the public key is stored by the online service.
 
+=== Authentication ceremony
+Authentication happens locally on the authenticator and is verified by the online service. First, a user will make a request to login. The online service will send the client some data, which includes a challenge. Next, the client requests cryptographic proof of user authentication. With the user's consent, the authenticator will return an assertion. This assertion provides evidence that the authenticator is in possession of the private key by signing the client data with the private key. The server then verifies the client data has not been tampered with and uses the public key to verify the signature.
```

Comment:
Subtle correction: the client data is not built by the RP, but by the client with input (the challenge) from the RP.

```suggestion
Authentication happens locally on the authenticator and is verified by the online service. First, a user will make a request to login. The online service will send the client a challenge, which the client wraps in its _client data_. Next, the client requests cryptographic proof of user authentication. With the user's consent, the authenticator will return an assertion. This assertion provides evidence that the authenticator is in possession of the private key by signing the client data with the private key. The server then verifies that the client data contains the correct challenge and uses the public key to verify the signature.
```